extensions/workouts/src/calculator.tsx

```diff
@@ -0,0 +1,146 @@
+import { Form } from "@raycast/api";
+import { useCachedState } from "@raycast/utils";
+import { useState, useEffect } from "react";
+import { CalculationMode, distancePresets, PaceCalculatorForm, type DistancePreset } from "./constants";
+
+export default function Calculator() {
+  const [calculatedPace, setCalculatedPace] = useState<string>("00:00:00");
+  const [calculatedTime, setCalculatedTime] = useState<string>("00:00:00");
+  const [distance, setDistance] = useState<string>("");
+  const [time, setTime] = useState<string>("");
+  const [pace, setPace] = useState<string>("");
+  const [mode, setMode] = useState<CalculationMode>("pace");
+  const [unit, setUnit] = useCachedState<"km" | "mi">("unit", "km");
+  const [selectedPreset, setSelectedPreset] = useState<string>("");
+
+  useEffect(() => {
```

Comment:
This useEffect seems unnessecary, the calculatedPace/calculatedTime state could be derived directly from the others without needing it's own state

Reply:
Should be better in https://github.com/raycast/extensions/pull/16338/commits/0c2408746b25f04d99da17a96c4f63dda9fc376b